### PR TITLE
Feature/refactor managers

### PR DESCRIPTION
**Pull request by JorxSS**
Current Mangers are refactored now.
 * Methods from World Manager to all get current Player 1 and Player 2. (Independent from Game Object, just based on the player's selection)
* All world event logic are currently managed in the Event Manager.
* All the current scripts uses the new method to get player 1 and 2.

When Debug Mode enable:
* Teleport to pre-set zones. (In order to do that, you need to make a DebugSystem's empty game object child, and place it wherever you want)
* Invincible mode now have a damage bonification against enemies.
* Spawn Mushdooms in front of you.

### CODE REVIEW

File: Scripts/src/Script/LightBall.cpp [hunk 1]
```diff
@@ -43,13 +43,13 @@ void LightBall::Awake()
...
+	//player1 = App->scene->GetGameObjectByName("Player1");
+	//ComponentScript* player1_component = player1->GetComponentScript("PlayerController");
+	//player1_controller = static_cast<PlayerController*>(player1_component->script);
 
-	player2 = App->scene->GetGameObjectByName("Player2");
-	ComponentScript* player2_component = player2->GetComponentScript("PlayerController");
-	player2_controller = static_cast<PlayerController*>(player2_component->script);
+	//player2 = App->scene->GetGameObjectByName("Player2");
+	//ComponentScript* player2_component = player2->GetComponentScript("PlayerController");
+	//player2_controller = static_cast<PlayerController*>(player2_component->script);
```
Comment (Dartemiss) [inline Scripts/src/Script/LightBall.cpp:52]: Why not erase all this code?

File: Scripts/src/Script/LightBall.cpp [hunk 2]
```diff
@@ -58,29 +58,35 @@ void LightBall::Awake()
...
+		//}
+		//else
+		//{
+		//	current_player = player2;
+		//	next_target = player2;
+		//	current_player_controller = player2_controller;
+		//	next_target_controller = player2_controller;
+		//}
+		current_player = next_target = player_1->player_go;
+		current_player_controller = next_target_controller = player_1->player_controller;
```
Comment (Dartemiss) [inline Scripts/src/Script/LightBall.cpp:81]: Why not erase this code?

File: Scripts/src/Script/PlayerAttack.cpp [hunk 3]
```diff
@@ -100,7 +106,14 @@ void PlayerAttack::ComputeCollisions() const
...
 			}
-			enemy->TakeDamage(current_damage_power);
+			if(debug->is_player_invincible)
+			{
+				enemy->TakeDamage(current_damage_power * debug_damage);
+			}
+			else
+			{
+				enemy->TakeDamage(current_damage_power);
+			}
```
Comment (Dartemiss) [inline Scripts/src/Script/PlayerAttack.cpp:116]: This stills nasty :O, you should just change the value of debug_damage to 300 or to 1.0f and keep only one clean and beautiful enemy->TakeDamage(current_damage_power * debug_damage);

Comment (JorxSS) [reply Scripts/src/Script/PlayerAttack.cpp]: fixed

File: Scripts/src/Script/LightBall.cpp (hunk 2 above)
Comment (JorxSS) [reply Scripts/src/Script/LightBall.cpp]: done

File: Scripts/src/Script/LightBall.cpp (hunk 1 above)
Comment (JorxSS) [reply Scripts/src/Script/LightBall.cpp]: done

**Pull request merged**